csrc/ir/utils.h

```diff
@@ -505,4 +495,24 @@ bool isTensorSize(const Val* val);
 //! Check if a Val is a tensor stride;
 bool isTensorStride(const Val* val);
 
+//! This is a simple template function that returns a vector of the given op
+//! type. Directly implemented in the header file to avoid explicit
+//! instantiation.
+template <typename OpType>
+std::vector<OpType*> getOpsOfType(Fusion* fusion) {
+  std::vector<OpType*> ops;
+  for (auto expr : fusion->exprs()) {
+    if (expr->isA<OpType>()) {
+      ops.push_back(expr->as<OpType>());
+    }
+  }
+  return ops;
+}
```

Comment:
Should we do this?
```C++
template <typename ... OpTypes>
auto getOpsOfType(Fusion* fusion) {
  using FirstOpType = std::tuple_element_t<0, std::tuple<Types...>>;
  using ExprType = std:: conditional_t<sizeof...(Types) == 1, FirstOpType, Expr>;
  std::vector<ExprType*> ops;
  for (auto expr : fusion->exprs()) {
    if (expr->isOneOf<OpTypes...>()) {
      ops.push_back(expr->as<ExprType>());
    }
  }
  return ops;
}
```
This way, we can remove `getAllTypesOfReductionOps` as well.

Reply:
I considered variadic template. But it requires the caller knows all types of reduction ops, e.g.
`auto reduction_ops = getOpsOfType<ReductionOp, GroupedReductionOp, WelfordOp>(fusion)`. If this is not a problem, we can remove `getAllTypesOfReductionOps`

Reply:
Actually I can rewrite the `getAllTypesOfReductionOps` function using the variadic template. Thanks for the suggestion!

Reply:
This makes sense. Thanks for the clean up!